gc/base/MemoryPoolAddressOrderedList.hpp

```diff
@@ -146,13 +160,94 @@ class MM_MemoryPoolAddressOrderedList : public MM_MemoryPoolAddressOrderedListBa
 	virtual uintptr_t releaseFreeMemoryPages(MM_EnvironmentBase* env);
 #endif
 
+	/* for Balanced GC only */
+	/**
+	 * Increase the scannable/non-scannable estimate for the receiver by the specified amount
+	 * @param scannableBytes the number of bytes to increase for scannable objects
+	 * @param non-scannableBytes the number of bytes to increase for scannable objects
+	 */
+	MMINLINE void incrementScannableBytes(uintptr_t scannableBytes, uintptr_t nonScannableBytes) { _scannableBytes += scannableBytes; _nonScannableBytes += nonScannableBytes; }
+	/**
+	 * @return the recorded estimate of scannable in the receiver
+	 */
+	MMINLINE uintptr_t getScannableBytes() { return _scannableBytes; }
+	/**
+	 * @return the recorded estimate of non-scannable in the receiver
+	 */
+	MMINLINE uintptr_t getNonScannableBytes() { return _nonScannableBytes; }
+
+	/**
+	 * remove a free entry from freelist
+	 */
+	void removeFromFreeList(void *addrBase, void *addrTop, MM_HeapLinkedFreeHeader *previousFreeEntry, MM_HeapLinkedFreeHeader *nextFreeEntry)
+	{
+		bool const compressed = compressObjectReferences();
+		uintptr_t freeEntrySize = ((uintptr_t)addrTop) - ((uintptr_t)addrBase);
+		MM_HeapLinkedFreeHeader::fillWithHoles(addrBase, freeEntrySize, compressed);
+		if (previousFreeEntry) {
+			previousFreeEntry->setNext(nextFreeEntry, compressed);
+		}else {
+			_heapFreeList = nextFreeEntry;
+		}
+	}
+
+	void fillWithHoles(void *addrBase, void *addrTop)
+	{
+		bool const compressed = compressObjectReferences();
+		MM_HeapLinkedFreeHeader::fillWithHoles(addrBase, ((uintptr_t)addrTop) - ((uintptr_t)addrBase), compressed);
+	}
+
+	MMINLINE uintptr_t getAdjustedbytesForCardAlignment()
+	{
+		return _adjustedbytesForCardAlignment;
+	}
+
+	MMINLINE void setAdjustedbytesForCardAlignment(uintptr_t adjustedbytes, bool needSync)
+	{
+		if (needSync) {
+			MM_AtomicOperations::add(&(_adjustedbytesForCardAlignment), adjustedbytes);
+		} else {
+			_adjustedbytesForCardAlignment += adjustedbytes;
+		}
+	}
+
+	MMINLINE void initialFirstUnAlignmentFreeEntry()
+	{
+		_firstUnalignedFreeEntry = (NULL == _heapFreeList) ? FREE_ENTRY_END : _heapFreeList;
+		_prevFirstUnalignedFreeEntry =  FREE_ENTRY_END;
+	}
+
+	MMINLINE void resetFirstUnAlignmentFreeEntry()
+	{
+		_firstUnalignedFreeEntry =  FREE_ENTRY_END;
+		_prevFirstUnalignedFreeEntry =  FREE_ENTRY_END;
+	}
```

Comment:
These 2 helpers don't seem to be used (although they possible can). Either remove them, or if you do end up using them, rename to

....FirstUnalignedFreeEntry()